src/containers/Chat/ChatMessages/AddAttachment/AddAttachment.test.tsx

```diff
@@ -117,3 +123,46 @@ test('show warnings if attachment type is sticker', () => {
   const helpText = screen.getByText('Animated stickers are not supported.');
   expect(helpText).toBeInTheDocument();
 });
+
+test('should get error notification if uploading media fails', async () => {
+  const defaultProps = {
+    setAttachment,
+    setAttachmentURL,
+    setAttachmentAdded,
+    setAttachmentType,
+    attachmentURL: '',
+    attachmentType: 'IMAGE',
+    uploadPermission: true,
+  };
+  render(<MockedProvider mocks={[uploadMediaErrorMock]}>
+    <AddAttachment {...defaultProps} />
+  </MockedProvider>)
+
+  const file = { name: 'photo.png' };
+
+  fireEvent.change(screen.getByTestId('uploadFile'), { target: { files: [file] } });
+
+  await waitFor(() => {
+    expect(setNotification).toHaveReturnedWith('warning');
+  });
+})
+
+test('successful media submission', async () => {
+  const responseData = { data: { is_valid: true } };
+  act(() => {
+    mockedAxios.get.mockImplementationOnce(() => Promise.resolve(responseData));
+  });
+  render(addAttachment('IMAGE', 'https://glific.com'));
+
+  // delay to let url validation complete
+  await waitFor(() => new Promise(res => setTimeout(res, 50)));
```

Comment:
await waitfor already adds a delay by waiting for the expected result. We don't need to use Promise and setTimeout to add delays

Reply:
> await waitfor already adds a delay by waiting for the expected result. We don't need to use Promise and setTimeout to add delays

The reason I added it is because I am not expecting URL validation to complete with `await waitFor(() => {
    fireEvent.click(screen.getByTestId('ok-button'));
  })`.
  That waitFor just clicks the ok-button as soon as its found and moves ahead, but I only want the ok-button to be clicked once the url is validated.
  The test fails without the added delay.